### PR TITLE
refactor(ui5-dialog): convert to TS

### PR DESCRIPTION
**Pull request by gmkv**

### CODE REVIEW

File: packages/main/src/Dialog.ts [hunk 1]
```diff
@@ -210,7 +91,137 @@ const metadata = {
...
+	/**
+	 * @private
+	 */
+	@property({ type: Boolean })
+	onDesktop!: boolean;
+
+	_screenResizeHandler!: () => void;
+	_dragMouseMoveHandler!: (e: MouseEvent) => void;
+	_dragMouseUpHandler!: (e: MouseEvent) => void;
+	_resizeMouseMoveHandler!: (e: MouseEvent) => void;
```
Comment (ilhan007) [inline packages/main/src/Dialog.ts:186]: this was not discussed, but members initialised in the constructor can be declared with just ":" without "!"
_screenResizeHandler: () => void;
_dragMouseMoveHandler: (e: MouseEvent) => void;

File: packages/main/src/Dialog.ts [hunk 2]
```diff
@@ -210,7 +91,137 @@ const metadata = {
...
+
+	/**
+	 * Defines the header HTML Element.
+	 * <br><br>
+	 * <b>Note:</b> If <code>header</code> slot is provided, the labelling of the dialog is a responsibility of the application developer.
+	 * <code>accessibleName</code> should be used.
+	 *
+	 * @type {HTMLElement[]}
+	 * @slot
+	 * @public
```
Comment (ilhan007) [inline packages/main/src/Dialog.ts:210]: name for the slots is also required

Comment (ilhan007) [reply packages/main/src/Dialog.ts:210]: I meant @name sap.ui.webc.main.Dialog.prototype.header
and
sap.ui.webc.main.Dialog.prototype.footer

sorry for not clarifying it

Comment (gmkv) [reply packages/main/src/Dialog.ts:210]: Thanks, i fixed it now.

**Review by ilhan007 — APPROVED** [review]

**Pull request merged**